src/osp/drawing_gl/FullscreenTriShader.cpp

```diff
@@ -32,15 +32,72 @@
 #include <Corrade/Containers/Iterable.h>  // for Containers::Iterable
 #include <Corrade/Containers/Reference.h>
 
+#include <filesystem>
+#include <string>
+
+#if defined(_WIN32) || defined(_WIN64)
+    #include <windows.h>
+#elif defined(__APPLE__)
+    #include <mach-o/dyld.h>
+    #include <limits.h>
+#elif defined(__linux__)
+    #include <unistd.h>
+    #include <limits.h>
+#endif
+
 using namespace osp;
 using namespace Magnum;
 
+std::string getAppPath() 
+{
+    #if defined(_WIN32) || defined(_WIN64)
+
+    char path[MAX_PATH];
+    HMODULE hModule = GetModuleHandle(NULL);
+    GetModuleFileName(hModule, path, MAX_PATH);
+    return std::string(path);
+
+    #elif defined(__APPLE__)
+
+    char path[PATH_MAX];
+    uint32_t size = sizeof(path);
+    if (_NSGetExecutablePath(path, &size) == 0) 
+    {
+        return std::string(path);
+    } 
+    else 
+    {
+            return std::string();
+    }
+
+    #elif defined(__linux__)
+
+    char path[PATH_MAX];
+    ssize_t count = readlink("/proc/self/exe", path, PATH_MAX);
+    if (count != -1) 
+    {
+        return std::string(path);
+    } 
+    else 
+    {
+        return std::string();
+    }
+
+    #endif
+}
+
 FullscreenTriShader::FullscreenTriShader()
 {
     GL::Shader vert{GL::Version::GL430, GL::Shader::Type::Vertex};
     GL::Shader frag{GL::Version::GL430, GL::Shader::Type::Fragment};
-    vert.addFile("OSPData/adera/Shaders/FullscreenTri.vert");
-    frag.addFile("OSPData/adera/Shaders/FullscreenTri.frag");
+    std::string const appPath = getAppPath();
+    if ( ! std::filesystem::exists(appPath + "OSPData/adera/Shaders/FullscreenTri.vert")
+    || ! std::filesystem::exists(appPath + "OSPData/adera/Shaders/FullscreenTri.frag"))
+    {
+        return;
```

Comment:
This should probably log an error instead of a silent return

Reply:
iostream was not included hence I did not add any cerr or cout

Reply:
Oh I see ! There is a OSP_LOG_ERROR (or any other level of log) macro defined in osp/util/logging.h

Reply:
Oh thanks!